components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/UnifiedClaimMetadataManager.java

```diff
@@ -342,8 +349,12 @@ public Optional<ExternalClaim> getExternalClaim(String externalClaimDialectURI,
         Optional<ExternalClaim> externalClaimInSystem = this.systemDefaultClaimMetadataManager.getExternalClaim(
                 externalClaimDialectURI, claimURI, tenantId);
         if (externalClaimInSystem.isPresent()) {
-            externalClaimInSystem.get().setClaimProperty(ClaimConstants.IS_SYSTEM_CLAIM, Boolean.TRUE.toString());
-            return externalClaimInSystem;
+            boolean isMappedLocalClaimInDB = externalClaimsInDB.stream().anyMatch(claim ->
+                    claim.getMappedLocalClaim().equals(externalClaimInSystem.get().getMappedLocalClaim()));
+            if (!isMappedLocalClaimInDB) {
+                externalClaimInSystem.get().setClaimProperty(ClaimConstants.IS_SYSTEM_CLAIM, Boolean.TRUE.toString());
```

Comment:
Shall we use a helper method for this to reduce the redundancy and improve readability as we are doing this operation in multiple places?

```
    private void markAsSystemClaim(ExternalClaim externalClaim) {
        externalClaim.setClaimProperty(ClaimConstants.IS_SYSTEM_CLAIM, Boolean.TRUE.toString());
    } 
```

Reply:
Fixed with commit acef81c7d06f394c23ff549f4f559641436207e9. Added new private method private void markAsSystemClaim(Claim claim) since this property is set for both external and local claims.